### PR TITLE
Updating the documentation for contributing functions in esql.

### PR DESCRIPTION
**Pull request by gheorghepucea**
Added some small corrections and clarifications to the documentation on how to contribute functions in esql.

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java [hunk 1]
```diff
@@ -129,13 +130,14 @@
...
- *         Make sure to commit them and reference them in your doc file.
+ *         Make sure to commit them and reference them in your doc file. There are plenty of examples on how
+ *         to reference those files e.g. {@code docs/reference/esql/functions/sin.asciidoc}.
  *     </li>
  *     <li>
  *          Build the docs by cloning the <a href="https://github.com/elastic/docs">docs repo</a>
  *          and running:
  *          <pre>{@code
- * ../docs/build_docs --doc docs/reference/index.asciidoc --resource x-pack/docs/ --open --chunk 1
+ * ../docs/build_docs --doc docs/reference/index.asciidoc --resource docs/ --open --chunk 1
```
Comment (nik9000) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java:140]: The original line is correct here I believe.

**Review by nik9000 — COMMENTED** [review]
thanks for updating this! I think your change in the command for how to build the docs is off, but otherwise it look right to me!

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java (hunk 1 above)
Comment (gheorghepucea) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java:140]:
```
➜  elasticsearch git:(update-docu) ../docs/build_docs --doc docs/reference/index.asciidoc --resource x-pack/docs/ --open --chunk 1
Can't find --resource /Users/${user}/git-code/elasticsearch/x-pack/docs
```
It very well could be, for me it's throwing the above error, what did I do wrong?

Comment (nik9000) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java:140]: Weird. When I copy that command precisely as you type it it works locally. But when I look in the x-pack/docs directory it's just old garbage, like we removed the directory long ago but my laptop still has it. I think, then, that you can remove the `--resource` param entirely. That works locally for me. How about you?

Comment (gheorghepucea) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java]: Yep, removing the parameter also does the trick. Less parameters, less confusion :) I've updated it.

Comment (nik9000) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java]: Turns out it's best to just drop the `--resource`.

**Review by nik9000 — APPROVED** [review]

**Pull request merged**